### PR TITLE
check pBaseTx for null

### PR DESCRIPTION
**Pull request by dudebing99**

### CODE REVIEW

File: src/rpc/rpctx.cpp [hunk 1]
```diff
@@ -2525,11 +2525,13 @@ Value sigstr(const Array& params, bool fHelp) {
...
 		throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "to address Invalid  ");
 	}
+	Object obj;
 	CDataStream stream(vch, SER_DISK, CLIENT_VERSION);
 	std::shared_ptr<CBaseTransaction> pBaseTx;
 	stream >> pBaseTx;
-	CAccountViewCache view(*pAccountViewTip, true);
-	Object obj;
+	if (!pBaseTx.get()) {
+		return obj;
```
Comment (tigerich) [inline src/rpc/rpctx.cpp]: obj not assigned at all

Comment (dudebing99) [reply src/rpc/rpctx.cpp]: Once the input `str` can not be decoded, it's reasonable to return a null object.

Comment (dudebing99) [reply src/rpc/rpctx.cpp]: RYI, see `Object GetTxDetailJSON(const uint256& txhash)`.

Comment (tigerich) [reply src/rpc/rpctx.cpp]: then it'd be better to "return null" directly.

Comment (dudebing99) [reply src/rpc/rpctx.cpp]: Done

**Pull request merged**